### DISCUSSION TITLE
[eventscripter] re-write metaData when a media file finish

### DISCUSSION THREAD

**Discussion by jonlepage** [mgpai/resources#3] [Q&A] [answered]
Sup @mgpai , sorry to bother you again, i experimenting a little program for rewrite metaData of video media downloaded.
Related to last prost, i need a strong solutions.
I chosen to inject `<jd:prop:categories_comma_separated>`  to meta files `comment` , it is also indexed by the search engine from OS.

Everything work fine for now, am just have issue with jdownloader, i have some dificulty related to the env and doc.
Here my node program. (is working fine, no need hack there)
For now , i want pass 2 args from jdownloader. 
* `_filePath` of the finished video file
* `_comment`  (fake comment added from jdownload)
```js
const [np, ap, _filePath, _comment, ...batchPath] = process.argv;
const exiftool = require("exiftool-vendored").exiftool;
if (batchPath && batchPath.length) {
  //TODO: allow pass multiple path for batching process
  console.log("❌ sry, 1 argument can pass for now!, remove others ");
  process.exit(1);
} else {
  exiftool
    .write(_filePath, {
      /**Category or keywords are Microsoft Xtra Tags
       * not work fine and seem make currpte file
       * @see https://exiftool.org/TagNames/Microsoft.html#Xtra
       */
      // Category: "test1, test2",
      // Title: "",
      Comment: _comment,
    })
    .then(function (e) {
      console.log("🟩 metaData modified with sucess!", e);
      process.exit(0);
    })
    .catch(function (e) {
      console.log("❌ error when write metaData exif");
      process.exit(1);
    });
}
```

---

And here the eventScript, for now, am not sure if all ok, and also, i cant log file debug ! all is undefined or give me weird thing! 
```js
if (myDownloadLink?.getUrl() === 'pornhub.com') { // Proceed only for pornHub collections
    const _comment = myCrawledLink.getComment(); //  add metacomment from a packagescript setup.
    const _filePath = myDownloadLink.getFilePath(); // TODO: found method to get filePath dir ? : ex:"C://Video//Download//filename.mp4"
    const _programPath = "C:\\Users\\InformatiqueLepage\\Documents\\Dev\\repo\\JdownloaderMetaDataFFmpeg\\index.js";

    try {
        //writeFile("C:\\Users\\InformatiqueLepage\\Desktop\\shitLogTest.txt", JSON.stringify(myDownloadLink,null,'\t'), true);
        //callSync("node.exe","C:\\Users\\InformatiqueLepage\\Documents\\Dev\\repo\\JdownloaderMetaDataFFmpeg\\index.js",filePath,comment);
        callAsync(function(exitCode, stdOut, errOut) {
            alert(errOut);
        }, "node.exe", `${programPath} ${filePath} ${comment}`);
    } catch (e) {

         console.error('something wrong'); // need found the jdownload terminal !?
        alert(e)
    }
}
```

what's am doing wrong and how to proceed fine ? 
are you able to throw me a sample that's you think will work ?
I want execute this each time a file finish.
thanks again.

**Comment by mgpai** [comment, marked as answer]
You can execute external programs directly from eventscripter. Here is an example script:
https://board.jdownloader.org/showpost.php?p=400706&postcount=106

**Reply by jonlepage** [reply]
thank , this will help a lot., i take a look and try.

**Reply by jonlepage** [reply]
> You can execute external programs directly from eventscripter. Here is an example script:
> https://board.jdownloader.org/showpost.php?p=400706&postcount=106

Your awesome , it work fine, thanks for the sample.

Last  questions, why it alway poke me this message   ?
![image](https://user-images.githubusercontent.com/24865815/109999271-0d255100-7ce0-11eb-985e-9d475815b965.png)

I check **dont show me again,** and it always promp again ! ?

**Reply by mgpai** [reply]
Permission is also required each/every time the script is modified. Following method can be used to disable it.
```js
disablePermissionChecks();
```
